unblob/handlers/archive/cpio.py

```diff
@@ -53,6 +80,20 @@ def calculate_chunk(
                     offset += self._pad_content(header, c_filesize, c_namesize)
                     break
 
+            c_mode = self._calculate_mode(header)
+
+            file_type = c_mode & 0o770000
+            sticky_bit = c_mode & 0o7000
+            permission = c_mode & 0o777
+
+            # heuristics 3: check mode field
+            if not (
+                any([file_type == ft for ft in C_FILE_TYPES])
+                and any([sticky_bit == sb for sb in C_STICKY_BITS])
+                and permission <= 0o777
+            ):
+                return
```

Comment:
Maybe I miss something, but I think this fragment could be simplified to:

```suggestion
            is_valid = file_type in C_FILE_TYPES and sticky_bit in C_STICKY_BITS
            if not is_valid:
                return
```

Note, that permission related condition is not there, as I do not see how the permission check could ever fail.